test/sugar.cmake

```diff
@@ -0,0 +1,12 @@
+sugar_files(TEST_SOURCES
```

Comment:
This just looks like a duplicate of `test/CMakeLists.txt`, violating DRY (Don't Repeat Yourself). If you actually need to duplicate things with hunter/sugar to be able to build without it then I think you should just build without it. This just seems to double required maintenance effort.

Reply:
In my projects (globally deployed services for very large companies with massive user bases) I always use hunter because I value the reliability, good integration with CI and automatic cross-compilation.

However, I am aware that not everyone will be persuaded of the benefits on first sight (usually the moment someone has actually used Hunter, they never stop using it).

So I thought I'd put equivalent code so that people could use it both ways.

I am a busy person so for me to contribute to a project which has dependencies (even just one), it is helpful to have automatic dependency management.

I don't keep a copy of the boost libraries in my usr/lib because I am often cross-compiling against a number of different systems and versions of c++ and I can't risk cmake etc picking up the wrong version.

Hunter solves that too.

However, this is your project. If the commit is not welcome, no problem. I'll just play on my fork and just contribute code changes.

Another solution might be to have a separate subdirectory for (vanilla) cmake and one for hunterised cmake.

If you ever want this package to be adopted as a hunter package (you do). Then a hunterised version of the cmakelists.txt file will be made in the hunter packaging fork in any case, because the dependency on boost needs to be managed.

Thanks for reading the commit.

R

Reply:
From your description I get the impression that Hunter, while it uses CMake, is in practice a completely different build system that isn't interchangeable with plain CMake. I had hoped it would be able to just, at CMake-generation-time, walk all targets, extract its properties, and use those to produce an exportable "package". If it requires interfacing with CMake through it's own abstraction layer that seems rather unfortunate because that's a very invasive approach.

Reply:
It's not completely different.

You can use Sugar without deploying it from hunter, it's just that hunter supports it out of the box.

Sugar is about source file management (something that cmake is bad at - since source file properties don't migrate out of sub_directories).

Hunter is about dependency management.

You can simply ignore all hunter code in a cmake project by setting the cache variable HUNTER_ENABLED=NO

As long as your packages can be found, it will all still work.